lib/cucumber/support_code/step_definition.js

```diff
@@ -106,19 +106,30 @@ function StepDefinition(pattern, options, code) {
 
       Cucumber.Util.Exception.registerUncaughtExceptionHandler(handleException, stepDomain);
 
+      var validCodeLengths = self.validCodeLengths(parameters);
+      if (validCodeLengths.indexOf(code.length) === -1) {
+        var error = 'definition has ' + code.length + ' arguments' +
```

Comment:
I went with definition as this also applies to hooks, but I'll update it to say the proper thing dependent on the context